src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisConfigDrawer.tsx

```diff
@@ -0,0 +1,191 @@
+ * licensable by each licensor hereunder covering either (i) the unmodified
+ * Software as contributed to or provided by such licensor, or (ii) the Larger
+ * Works (as defined below), to deal in both
+ *
+ * (a) the Software, and
+ * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ * one is included with the Software (each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import { automatedAnalysisConfigToRecordingAttributes, RecordingAttributes } from '@app/Shared/Services/Api.service';
+import { ServiceContext } from '@app/Shared/Services/Services';
+import { useSubscriptions } from '@app/utils/useSubscriptions';
+import {
+  Drawer,
+  DrawerActions,
+  DrawerCloseButton,
+  DrawerContent,
+  DrawerContentBody,
+  DrawerHead,
+  DrawerPanelContent,
+  Dropdown,
+  DropdownItem,
+  DropdownToggle,
+  DropdownToggleAction,
+  Level,
+  LevelItem,
+} from '@patternfly/react-core';
+import { CogIcon, PlusCircleIcon } from '@patternfly/react-icons';
+import * as React from 'react';
+import { first } from 'rxjs';
+import { AutomatedAnalysisConfigForm } from './AutomatedAnalysisConfigForm';
+
+interface AutomatedAnalysisConfigDrawerProps {
+  drawerContent: React.ReactNode;
+  onCreate: () => void;
+}
+
+export const AutomatedAnalysisConfigDrawer: React.FunctionComponent<AutomatedAnalysisConfigDrawerProps> = (props) => {
+  const context = React.useContext(ServiceContext);
+  const addSubscription = useSubscriptions();
+
+  const [isExpanded, setIsExpanded] = React.useState(false);
+  const [isDropdownOpen, setIsDropdownOpen] = React.useState(false);
+  const drawerRef = React.useRef<HTMLDivElement>(null);
+
+  const onToggle = React.useCallback(
+    (selected) => {
+      setIsDropdownOpen(selected);
+    },
+    [setIsDropdownOpen]
+  );
+
+  const onDropdownFocus = () => {
```

Comment:
Could you explain that is this for? I cant find an element with this id tho?

Reply:
Was it meant to focus on the toggle element? So it should be `automated-analysis-recording-config-toggle` right?

Reply:
But this function is called after selecting. At this point, we already set focus on the drawer expanded content. So, I am not sure its a good idea to switch focus to dropdown.

Reply:
Sorry, I was using the wrong id.